packages/core/src/metrics/extract-navigation-sample.ts

```diff
@@ -94,28 +97,46 @@ function findPhases(events: EventModel[], markers: Marker[]): PhaseSample[] {
       phaseEvents.push(markEvent);
     }
   }
-  let paintEvent: CompleteEventModel | undefined;
+  let paintEvent: CompleteEventModel | MarkEventModel | undefined;
   for (; eventIdx < events.length; eventIdx++) {
     const event = events[eventIdx];
-    if (event.isComplete() && event.name === 'Paint') {
+    if (
+      (event.isComplete() && event.name === 'Paint') ||
+      (event.isMark() && event.name === LCP_EVENT_NAME && traceEndAtLcp)
+    ) {
       paintEvent = event;
       break;
     }
   }
   if (!paintEvent) {
     throw new Error(
-      `Could not find Paint event in trace after last mark "${
+      `Could not find Paint event or ${LCP_EVENT_NAME} in trace after last mark "${
         markers[markers.length - 1].start
-      }"`
+      }. The last marker should trigger a paint event."`
     );
   }
   const phases: PhaseSample[] = [];
   const start = phaseEvents[0].start;
+  let end = phaseEvents[0].end;
+
   for (let i = 0; i < phaseEvents.length; i++) {
     const marker = markers[i];
     const markEvent = phaseEvents[i];
-    const end =
-      i + 1 < phaseEvents.length ? phaseEvents[i + 1].start : paintEvent.end;
+
+    if (i + 1 < phaseEvents.length) {
+      //skip LCP if user configure the LCP in the middle of marker list
+      if (isLCPEvent(markEvent.name)) {
```

Comment:
If we wanted to in the future genericize performance marks as markers this would have to be updated to check with a more generic function, correct?

Reply:
yes, for example, we can define enum that includes more performance marks.